### PR TITLE
Update to V7 and new app

### PR DESCRIPTION
**Pull request by wswebcreation**
This PR will upgrade the boilerplate to:

- V7
- TypeScript
- New App

Tests and docs will be migrated per tests:

- [x] `app.forms.spec.ts`
- [x] `app.login.spec.ts`
- [x] `app.navigation.spec.ts` => renamed to `app.tab.bar.navigation.spec.ts`
- [x] `app.swipe.spec.ts`
- [x] `app.webview.spec.ts`
- [x] `app.webview.xpath.spec.ts`
- [x] `browser.checkbox.spec.ts`
- [x] `browser.dynamic.spec.ts`
- [x] `browser.form.spec.ts`

New tests need to be added:
- [x] Biometrics tests =>`app.biometric.login.spec.ts`
- [x] Deep link tests =>`app.deep.link.navigation.spec.ts`
- [x] Swipe horizontal => `app.swipe.spec.ts`
- [x] Puzzle tests for drag and drop => `app.drag.and.drop.spec.ts`

I also need to add the following:
- [x] Better docs per test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,27 +13,29 @@ Boilerplate project to run Appium tests together with WebdriverIO for:
...
-> The releases can be found and downloaded [here](https://github.com/webdriverio/native-demo-app/releases).
-> Before running tests, please create a `./apps` directory, download the app and move the zip files into that directory
+> The releases can be found and downloaded [here](https://github.com/webdriverio/native-demo-app/releases), this boilerplate works with
+> version `0.4.0` or higher.
+> Before running tests, please create a `./apps` directory, download the apps and move the files into that directory
 
 > **Note:**
-> This boilerplate only handles local execution on 1 em/simulator at a time, not parallel execution. For more info about that Google on setting up a grid with Appium.
+> This boilerplate only handles local execution on 1 em/simulator at a time, not parallel execution. For more info about that Google on
+> setting up a grid with Appium.
```
Comment (christian-bromann) [inline README.md:22]:
```suggestion
> This boilerplate only handles local execution on 1 em/simulator at a time, not parallel execution. For more info about that Google on setting up a grid with Appium.
```

File: config/browserstack/wdio.android.bs.app.conf.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-const { config } = require('../wdio.shared.conf');
+// @ts-ignore
+const { config }:{config: WebdriverIO.Config} = require('../wdio.shared.conf');
```
Comment (christian-bromann) [inline config/browserstack/wdio.android.bs.app.conf.ts:2]: We can use `import` here too
```suggestion
import { config } from '../wdio.shared.conf':
```

File: config/saucelabs/wdio.android.emulators.app.conf.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
+// @ts-ignore
+const { config }:{config: WebdriverIO.Config} = require('./wdio.shared.sauce.conf');
```
Comment (christian-bromann) [inline config/saucelabs/wdio.android.emulators.app.conf.ts:2]: same

File: config/saucelabs/wdio.ios.simulators.app.conf.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
+// @ts-ignore
+const { config }:{config: WebdriverIO.Config} = require('./wdio.shared.sauce.conf');
```
Comment (christian-bromann) [inline config/saucelabs/wdio.ios.simulators.app.conf.ts:2]: same

File: config/wdio.android.browser.conf.ts [hunk 5]
```diff
@@ -1,10 +1,11 @@
-const { config } = require('./wdio.shared.conf');
+// @ts-ignore
+const { config }:{config: WebdriverIO.Config} = require('./wdio.shared.local.appium.conf');
```
Comment (christian-bromann) [inline config/wdio.android.browser.conf.ts:2]: same

File: config/wdio.shared.local.appium.conf.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+// @ts-ignore
+const { config }:{config: WebdriverIO.Config} = require('./wdio.shared.conf');
```
Comment (christian-bromann) [inline config/wdio.shared.local.appium.conf.ts:2]: same

**Review by christian-bromann — COMMENTED** [review]
Fantastic work, Wim!

Made a couple of comments, nothing critical. I wonder how much from this we could move into the WebdriverIO core to make life easier for folks not using this boilerplate.

File: tests/screenobjects/DragScreen.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    get dropC3(): WebdriverIO.Element {return $('~drop-c3');}
+    get dropR3(): WebdriverIO.Element {return $('~drop-r3');}
+    get renew(): WebdriverIO.Element {return $('~renew');}
+    get retry(): WebdriverIO.Element {return $('~button-Retry');}
+
+    /**
+     * Drag an element to a position.
+     */
+    dragElementTo(elementOne: WebdriverIO.Element, elementTwo: WebdriverIO.Element):void {
+        driver.touchPerform([
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply tests/screenobjects/DragScreen.ts:33]: Yep, something we can add. For now I'll keep it in here so everybody knows how to do it with the `touchPerform`

File: tests/screenobjects/AndroidSettings.ts [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    findAndroidElementByText(string:string):WebdriverIO.Element {
+        const selector = `android=new UiSelector().textContains("${string}")`;
+
+        return $(selector);
+    }
+
+    /**
+     * Wait and click on an element
+     */
+    waitAndClick(string: string): void{
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply tests/screenobjects/AndroidSettings.ts:38]: Will change that, thanks!

File: scripts/push_apps_to_sauce_storage.sh [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  -F "payload=@../apps/Android-NativeDemoApp-$VERSION.apk" \
+  -F name=wdio-demo-app-android.apk \
+  -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY"  \
+  'https://api.eu-central-1.saucelabs.com/v1/storage/upload'
+# iOS
+curl \
+  -F "payload=@../apps/iOS-Simulator-NativeDemoApp-$VERSION.app.zip" \
+  -F name=wdio-demo-app-ios.zip \
+  -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY"  \
+  'https://api.eu-central-1.saucelabs.com/v1/storage/upload'
```
[the remark this answers is not in the document]
Comment (wswebcreation) [reply scripts/push_apps_to_sauce_storage.sh:31]: Yeah, that could be useful

File: config/browserstack/wdio.android.bs.app.conf.ts (hunk 2 above)
Comment (wswebcreation) [reply config/browserstack/wdio.android.bs.app.conf.ts:2]: Updated them all, thans

File: .eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        es6: true,
+    },
+    globals: {
+        describe: true,
+        xdescribe: true,
+        fdescribe: true,
+        it: true,
+        xit: true,
+        fit: true,
+        expect: true,
```
Comment (christian-bromann) [inline .eslintrc.js:19]: I believe we can remove these when adding `jasmine: true` to `env`

File: .eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        $: true,
+        $$: true,
+        driver: true,
+        beforeEach: true,
+        browser: true,
```
Comment (christian-bromann) [inline .eslintrc.js:24]: How about adding the wdio eslint plugin: https://www.npmjs.com/package/eslint-plugin-wdio .. then we could remove these too

File: config/browserstack/wdio.android.bs.app.conf.ts (hunk 2 above)
Comment (christian-bromann) [reply config/browserstack/wdio.android.bs.app.conf.ts:2]: still see `require` in this file

File: config/saucelabs/wdio.shared.sauce.conf.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+config.key = process.env.SAUCE_ACCESS_KEY;
+// If you run your tests on Sauce Labs you can specify the region you want to run your tests
+// in via the `region` property. Available short handles for regions are `us` (default) and `eu`.
+// These regions are used for the Sauce Labs Virtual cloud and the Sauce Labs Real Device Cloud.
+// If you don't provide the region, it defaults to `us`.
+config.region = process.env.REGION || 'us';
+// Increase for real device support
+config.connectionRetryTimeout = 180000;
+
+export default config;
```
Comment (christian-bromann) [inline config/saucelabs/wdio.shared.sauce.conf.ts:28]:
```suggestion
exports.config = config;
```

File: config/wdio.shared.local.appium.conf.ts [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    ],
+]);
+//
+// =====================
+// Server Configurations
+// =====================
+//
+config.port = 4723;
+
+export default config;
```
Comment (christian-bromann) [inline config/wdio.shared.local.appium.conf.ts:30]:
```suggestion

exports.config = config;
```

File: scripts/push_apps_to_sauce_storage.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (christian-bromann) [reply scripts/push_apps_to_sauce_storage.sh:31]: mind adding an issue for it in the main repo?

File: tests/specs/browser.form.spec.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import FormPage from '../pageobjects/form.page';
+
+fdescribe('auth form', () => {
```
Comment (christian-bromann) [inline tests/specs/browser.form.spec.ts:3]: was this intentional?
```suggestion
describe('auth form', () => {
```

File: config/browserstack/wdio.android.bs.app.conf.ts (hunk 2 above)
Comment (wswebcreation) [reply config/browserstack/wdio.android.bs.app.conf.ts:2]: sorry, missed that one

File: tests/specs/browser.form.spec.ts (hunk 14 above)
Comment (wswebcreation) [reply tests/specs/browser.form.spec.ts:3]: Thanks, was not intentional

File: .eslintrc.js (hunk 10 above)
Comment (wswebcreation) [reply .eslintrc.js:19]: Done!

File: .eslintrc.js (hunk 11 above)
Comment (wswebcreation) [reply .eslintrc.js]: Done

File: scripts/push_apps_to_sauce_storage.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (wswebcreation) [reply scripts/push_apps_to_sauce_storage.sh:31]: Done

**Pull request merged**